etl/src/main/java/org/pih/hivmigration/etl/sql/LabResultMigrator.groovy

```diff
@@ -0,0 +1,193 @@
+                obs_datetime,
+                value_numeric,
+                value_text,
+                value_boolean,
+                vl_beyond_detectable_limit,
+                vl_detectable_lower_limit)
+            values (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+        ''', '''
+            select   e.patient_id as source_patient_id,
+                     e.ENCOUNTER_ID as source_encounter_id, 
+                     r.RESULT_ID as source_result_id,
+                     r.sample_id, 
+                     lower(t.name) as test_type,
+                     to_char(e.encounter_date, 'yyyy-mm-dd') as obs_datetime,
+                     value as value_numeric, 
+                     value_string as value_text,   
+                     case when (value_p='t') then ( 1)     
+                        else 0
+                        end as value_boolean,                
+                     case when (r.RESULT_UNDETECTABLE='t') then ( 1)     
+                        else 0
+                        end as vl_beyond_detectable_limit,                  
+                     r.RESULT_UNDETECTABLE_RANGE as vl_detectable_lower_limit
+            from     hiv_demographics_real d, hiv_encounters e, hiv_lab_results r, hiv_lab_tests t
+            where    d.patient_id = e.patient_id
+            and      e.encounter_id = r.encounter_id
+            and      r.lab_test_id = t.lab_test_id  
+            order by d.patient_id desc, e.encounter_id desc
+        ''')
+
+        executeMysql("Add UUIDs", "UPDATE hivmigration_lab_results SET uuid = uuid();")
+
+        executeMysql("Create stored procedures to add obs", '''
+            DROP PROCEDURE IF EXISTS create_tmp_obs_table;
+            DROP PROCEDURE IF EXISTS migrate_tmp_obs;
+            DELIMITER $$ ;
+            CREATE PROCEDURE create_tmp_obs_table()
+            BEGIN
+                DROP TABLE IF EXISTS tmp_obs;
+                CREATE TABLE tmp_obs (
+                    obs_id INT PRIMARY KEY AUTO_INCREMENT,
+                    obs_group_id INT,
+                    source_patient_id INT,
+                    source_encounter_id INT,
+                    concept_uuid CHAR(38),
+                    value_coded_uuid CHAR(38),
+                    value_datetime DATETIME,
+                    value_numeric DOUBLE,
+                    value_text TEXT
+                );
+            END $$
+            CREATE PROCEDURE migrate_tmp_obs()
+            BEGIN
+                INSERT INTO obs (
+                    obs_id, person_id, encounter_id, obs_group_id, obs_datetime, location_id, concept_id,
+                    value_coded, value_numeric, value_datetime, value_text, creator, date_created, voided, uuid
+                )
+                SELECT
+                    o.obs_id, p.person_id, e.encounter_id, o.obs_group_id, e.encounter_date, e.location_id, q.concept_id,
+                    a.concept_id, o.value_numeric, o.value_datetime, o.value_text, 1, e.date_created, 0, uuid()
```

Comment:
My guess is that the "uuid()" function here probably should be replaced with the uuid that is being populated on the hivmigration_lab_results table, and that this represents the uuid of the group.  Correct?